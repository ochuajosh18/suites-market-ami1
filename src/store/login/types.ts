import { AnyAction } from 'redux';


export interface LoginState {
    username: string;
    password: string;
    rememberMe: boolean;
    isForgotPassword: boolean;
    email: string;
    newPassword: string;
    confirmPassword: string;
    emailError: boolean;
    emailHelperText: string;
    token: any;
    userNameError: boolean;
    userNameHelperText: string;
    passwordError: boolean;
    passwordHelperText: string;
    isLoggedIn: boolean;
    uuid: string;
    forgotPasswordAt: string;
    user: any;
    loginLoading: boolean;
    forgotPasswordLoading: boolean;
    newPasswordLoading: boolean;
    forgotPasswordId: string;
    newPasswordRequestLoading: boolean;
    basicUsername: string;
    basicPassword: string;
    basicRememberMe: boolean;
    platform: string;
    forgotPasswordSuccess: boolean;
    newPasswordSuccess: boolean;
    newPasswordStrengthValid: boolean;
}

export const CHANGE_USERNAME = 'change_username';
export const CHANGE_PASSWORD = 'change_password';
export const CHANGE_REMEMBER_ME = 'change_remember_me';
export const CHANGE_FORGOT_PASSWORD = 'change_forgot_password';
export const CHANGE_EMAIL = 'change_email';
export const CHANGE_EMAIL_VALID = 'change_email_valid';
export const CHANGE_NEW_PASSWORD = 'change_new_password';
export const CHANGE_CONFIRM_PASSWORD = 'change_confirm_password';

export const IS_NEW_PASSWORD_EMPTY = 'is_new_password_empty';
export const IS_CONFIRM_PASSWORD_EMPTY = 'is_confirm_password_empty';
export const IS_EMAIL_EMPTY = 'is_email_empty';
export const IS_USERNAME_EMPTY = 'is_username_empty';
export const IS_PASSWORD_EMPTY = 'is_password_empty';
export const IS_USERNAME_VALID = 'is_username_valid';

export const AUTHENTICATE = 'authenticate';
export const AUTHENTICATE_WITHOUTH_REMEMBER_ME =
    'authenticate_withouth_remember_me';
export const FORGOT_PASSWORD = 'forgot_password';

export const RESET_STATE = 'reset_login_state';
export const REFRESH_TOKEN = 'refresh_token';
export const PASSWORD_EMPTY = 'password_empty';
export const CHANGE_NEW_PASSWORD_HELPER_TEXT = 'change_password_helper_text';
export const CHANGE_CONFIRM_PASSWORD_HELPER_TEXT =
    'change_confirm_password_helper_text';
export const CHANGE_NEW_PASSWORD_ERROR = 'change_new_password_error';
export const CHANGE_CONFIRM_PASSWORD_ERROR = 'change_confirm_password_error';

export const CHANGE_LOGIN_LOADER = 'change_login_loader';
export const CHANGE_FORGOT_PASSWORD_LOADER = 'change_forgot_password_loader';
export const CHANGE_NEW_PASSWORD_LOADER = 'change_new_password_loader';
export const SET_FORGOT_PASSWORD_ID = 'set_forgot_password_id';
export const CHANGE_NEW_PASSWORD_REQUEST_LOADING =
    'change_new_password_request_loading';

export const SET_USER = 'set_user';
export const SET_LOGIN_STATE = 'set_login_state';

export interface LoginStateInput {
    [name: string]: string | Array<string> | Boolean;
}

export interface inputEmailAction {
    type: typeof CHANGE_USERNAME;
    payload: string;
}

export interface inputPasswordAction {
    type: typeof CHANGE_PASSWORD;
    payload: string;
}

export interface changeRememberMeAction {
    type: typeof CHANGE_REMEMBER_ME;
    payload: boolean;
}

export interface onForgotPasswordClickAction {
    type: typeof CHANGE_FORGOT_PASSWORD;
    payload: boolean;
}

export interface onChangeEmailAction {
    type: typeof CHANGE_EMAIL;
    payload: string;
}

export interface onChangeNewPassword {
    type: typeof CHANGE_NEW_PASSWORD;
    payload: string;
}

export interface onChangeConfirmPassword {
    type: typeof CHANGE_CONFIRM_PASSWORD;
    payload: string;
}

export interface newPasswordEmptyAction {
    type: typeof IS_NEW_PASSWORD_EMPTY;
    payload: object;
}

export interface confirmPasswordEmptyAction {
    type: typeof IS_CONFIRM_PASSWORD_EMPTY;
    payload: object;
}

export interface emailEmptyAction {
    type: typeof IS_EMAIL_EMPTY;
    payload: object;
}

export interface authenticateAction {
    type: typeof AUTHENTICATE;
    payload: string;
}

export interface usernameEmptyAction {
    type: typeof IS_USERNAME_EMPTY;
    payload: object;
}

export interface passwordEmptyAction {
    type: typeof IS_PASSWORD_EMPTY;
    payload: object;
}

export interface isUserNameValidAction {
    type: typeof IS_USERNAME_VALID;
    payload: object;
}

export interface ResetLoginState {
    type: typeof RESET_STATE;
    payload: object;
}

export interface RefeshTokenAction {
    type: typeof REFRESH_TOKEN;
    payload: object;
}

export interface PasswordEmptyAction {
    type: typeof PASSWORD_EMPTY;
    payload: null;
}

export interface setUserAction {
    type: typeof SET_USER;
    payload: object;
}

export interface LoginStateInputAction {
    type: typeof SET_LOGIN_STATE;
    payload: LoginStateInput
}

export type LoginAction =
    | AnyAction
    | inputEmailAction
    | inputPasswordAction
    | changeRememberMeAction
    | onForgotPasswordClickAction
    | onChangeEmailAction
    | onChangeNewPassword
    | onChangeConfirmPassword
    | newPasswordEmptyAction
    | confirmPasswordEmptyAction
    | emailEmptyAction
    | authenticateAction
    | usernameEmptyAction
    | passwordEmptyAction
    | ResetLoginState
    | RefeshTokenAction
    | ResetLoginState
    | PasswordEmptyAction
    | setUserAction
    | LoginStateInputAction;
