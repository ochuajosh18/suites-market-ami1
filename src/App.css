html,
body,
.root {
    /* Global Styling */
    width: 100%;
    height: 100%;
    font-family: Arial, Helvetica, sans-serif;
}

#root {
    display: flex;
    height: 100%;
    justify-content: center;
    font-family: Arial, Helvetica, sans-serif;
}

.MuiBox-root .MuiTypography-root,
.MuiBox-root .MuiButton-root span,
.MuiButtonBase-root span,
.MuiGrid-root .MuiButtonBase-root,
.MuiGrid-root .MuiMenuItem-root,
.MuiTypography-root *,
button span:not(span.fas,span.fa) {
    font-family: Arial, Helvetica, sans-serif;
}

.fa, .far, .fas {
    font-family: "Font Awesome 5 Free", sans-serif !important;
}


.App {
    width: 100%;
    height: 100%;
}

.App-header {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    width: 100%;
    height: 100%;
}

.round-border1 .MuiOutlinedInput-root {
    border-radius: 4rem !important;
}

.round-border2 .MuiOutlinedInput-root {
    border-radius: 0.5rem !important;
}

.round-border1.MuiButtonBase-root {
    border-radius: 4rem !important;
}

.header-logo-container {
    display: flex;
    align-items: center;
    width: 300px;
    max-height: 64px;
    box-sizing: border-box;
}

header {
    padding: 0 !important;
    margin: 0 !important;
}

/* .header-logo-container img {
    height: auto !important;
    position: static !important;
    width: 100% !important;
    max-width: 12rem;
    padding-left: 3rem;
} */

.header-logo-container img {
    max-height: 48px;
    width: auto;
    padding-left: 32px;
}

.header-right-container {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    height: 100%;
    width: 100%;
    padding: 0 1rem;
    flex: 1;
    box-sizing: border-box;
}

.Logo-container {
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    height: 64px;
}

.Logo-container div {
    display: flex !important;
    align-items: center !important;
    padding-left: 2rem !important;
    width: 100% !important;
}

.Logo-container img {
    position: static !important;
    height: auto !important;
    max-width: 9rem;
}

.mainText span {
    font-family: Arial, Helvetica, sans-serif;
    font-size: '1.5rem';
}

.subItem span {
    margin-left: '2rem';
}

.subText span {
    opacity: 0.5;
}

.small-text {
    text-transform: none!important;
}

.gmap-label {
    position: relative;
    bottom: 24px
}

.navicon {
    width: 24px;
    height: auto;
    max-height: 24px;
}

.word-break-all {
    word-break: break-all;
}
