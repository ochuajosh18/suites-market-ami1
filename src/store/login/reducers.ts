import {
    LoginAction,
    LoginState,
    CHANGE_USERNAME,
    CHANGE_PASSWORD,
    CHANGE_REMEMBER_ME,
    CHANGE_FORGOT_PASSWORD,
    CHANGE_EMAIL,
    CHANGE_EMAIL_VALID,
    CHANGE_NEW_PASSWORD,
    CHANGE_CONFIRM_PASSWORD,
    IS_EMAIL_EMPTY,
    AUTHENTICATE,
    IS_USERNAME_EMPTY,
    IS_PASSWORD_EMPTY,
    IS_USERNAME_VALID,
    AUTHENTICATE_WITHOUTH_REMEMBER_ME,
    FORGOT_PASSWORD,
    SET_USER,
    CHANGE_LOGIN_LOADER,
    CHANGE_FORGOT_PASSWORD_LOADER,
    CHANGE_NEW_PASSWORD_LOADER,
    SET_FORGOT_PASSWORD_ID,
    RESET_STATE,
    REFRESH_TOKEN,
    CHANGE_NEW_PASSWORD_REQUEST_LOADING,
    SET_LOGIN_STATE,
} from './types';

import moment from 'moment';

const INITIAL_STATE: LoginState = {
    username: '',
    password: '',
    rememberMe: false,
    isForgotPassword: false,
    email: '',
    newPassword: '',
    confirmPassword: '',
    emailError: false,
    emailHelperText: '',
    token: {},
    userNameError: false,
    userNameHelperText: '',
    passwordError: false,
    passwordHelperText: '',
    isLoggedIn: false,
    uuid: '',
    forgotPasswordAt: '',
    user: {
        firstName: '',
        lastName: '',
    },
    loginLoading: false,
    forgotPasswordLoading: false,
    newPasswordLoading: false,
    forgotPasswordId: '',
    newPasswordRequestLoading: false,
    basicUsername: '',
    basicPassword: '',
    basicRememberMe: false,
    platform: 'Symphony Market',
    forgotPasswordSuccess: false,
    newPasswordSuccess: false,
    newPasswordStrengthValid: false
};

export default (
    state = INITIAL_STATE,
    action: LoginAction,
): LoginState => {
    switch (action.type) {
        case CHANGE_USERNAME:
            return { ...state, username: action.payload };
        case CHANGE_PASSWORD:
            return { ...state, password: action.payload };
        case CHANGE_REMEMBER_ME:
            return { ...state, rememberMe: action.payload };
        case CHANGE_FORGOT_PASSWORD:
            return { ...state, isForgotPassword: action.payload };
        case CHANGE_EMAIL:
            return { ...state, email: action.payload };
        case CHANGE_EMAIL_VALID:
            return {
                ...state,
                emailError: action.payload.emailError,
                emailHelperText: action.payload.emailHelperText,
            };
        case CHANGE_NEW_PASSWORD:
            return { ...state, newPassword: action.payload };
        case CHANGE_CONFIRM_PASSWORD:
            return { ...state, confirmPassword: action.payload };
        case IS_EMAIL_EMPTY:
            return {
                ...state,
                emailError: action.payload.emailError,
                emailHelperText: action.payload.emailHelperText,
            };
        case AUTHENTICATE_WITHOUTH_REMEMBER_ME:
            return {
                ...state,
                token: action.payload,
                username: '',
                password: '',
                rememberMe: false,
                isLoggedIn: true,
                user: action.payload.userDetails
            };
        case AUTHENTICATE:
            return { ...state, token: action.payload, isLoggedIn: true, user: action.payload.userDetails};
        case IS_USERNAME_EMPTY:
            return {
                ...state,
                userNameError: action.payload.userNameError,
                userNameHelperText: action.payload.userNameHelperText,
            };
        case IS_PASSWORD_EMPTY:
            return {
                ...state,
                passwordError: action.payload.passwordError,
                passwordHelperText: action.payload.passwordHelperText,
            };
        case IS_USERNAME_VALID:
            const { userNameError, userNameHelperText } = action.payload;
            return { ...state, userNameError, userNameHelperText };
        case FORGOT_PASSWORD:
            return {
                ...state,
                uuid: action.payload,
                forgotPasswordAt: moment().toISOString(),
            };
        case SET_USER:
            return { ...state, user: action.payload };
        case CHANGE_LOGIN_LOADER:
            return { ...state, loginLoading: action.payload };
        case CHANGE_FORGOT_PASSWORD_LOADER:
            return { ...state, forgotPasswordLoading: action.payload };
        case CHANGE_NEW_PASSWORD_LOADER:
            return { ...state, newPasswordLoading: action.payload };
        case SET_FORGOT_PASSWORD_ID:
            return { ...state, forgotPasswordId: action.payload };
        case RESET_STATE:
            return { 
                ...INITIAL_STATE, 
                basicUsername: state.basicRememberMe ? state.basicUsername : '', 
                rememberMe: state.rememberMe, 
                basicRememberMe: state.basicRememberMe,
                platform: state.platform
            };
        case REFRESH_TOKEN: {
            return {
                ...state,
                token: { ...state.token, token: { token: action.payload } },
            };
        }
        case CHANGE_NEW_PASSWORD_REQUEST_LOADING:
            return { ...state, newPasswordLoading: action.payload };
        case SET_LOGIN_STATE:
            return { ...state, ...action.payload };
        default:
            return state;
    }
}
