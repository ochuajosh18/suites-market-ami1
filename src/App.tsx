import React from 'react';
import { connect } from 'react-redux';
import { loadCSS } from 'fg-loadcss';
import Box from '@material-ui/core/Box';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect,
} from 'react-router-dom';

import './App.css';

import { AppState } from './store';
import { refreshToken, resetLoginState, setLoginState } from './store/login/actions';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import MomentUtils from '@date-io/moment';
import { LoadScript } from '@react-google-maps/api';
import { Libraries } from '@react-google-maps/api/dist/utils/make-load-script-url';

import AuthenticatedRoute from './modules/AuthenticatedRoute';
import Home from './components/Home/Home';
// import OrderView from './components/Order/OrderView';
import ProductReview from './components/Review/ProductReview';
import ProductRating from './components/Review/ProductRating';
import UserManagement from './components/Administration/Administration';
import { setHeaderText } from './store/system/actions';
import SymphonyDrawer from './components/symphony/SymphonyDrawer';
import SymphonyHeader from './components/symphony/SymphonyHeader';

// Basic Components
import CallTypeSettings from './components/Basic/CallTypeSettings/CallTypeSettings';
import Promotion from './components/Basic/Promotion/Promotion';
import News from './components/Basic/News/News';
import Report from './components/sales/report/Report';
import AccessibilitySettings from './components/Basic/AccessibilitySettings/AccessibilitySettings';
import ApplicationSettings from './components/Basic/ApplicationSettings/ApplicationSettings';

// revamped
import SymphonyLogin from './components/symphony/SymphonyLogin';
import SymphonyForgotPassword from './components/symphony/SymphonyForgotPassword';
import SymphonyNewPassword from './components/symphony/SymphonyNewPassword';
import Role from './components/symphony/accessibility/Role';
import Fields from './components/symphony/fields/Fields';

// revamped market
import MarketProduct from './components/market/product/Product';
import MarketProductView from './components/market/product/ProductView';
import Category from './components/market/category/Category';
import PromoBanner from './components/market/promobanner/PromoBanner';
import PromoBannerView from './components/market/promobanner/PromoBannerView';
import Accessibility from './components/symphony/accessibility/Accessibility';
import Vendor from './components/market/usermanagement/Vendor';
import VendorView from './components/market/usermanagement/VendorView';
import Customer from './components/market/usermanagement/Customer';
import CustomerView from './components/market/usermanagement/CustomerView';
import AboutUs from './components/market/library/AboutUs';
import HelpDesk from './components/market/library/HelpDesk';
import Faqs from './components/market/library/Faq';
import UserAccount from './components/market/usermanagement/UserAccount';
import Order from './components/market/ordermanagement/Order';
import OrderView from './components/market/ordermanagement/OrderView';
import OrderHistory from './components/market/ordermanagement/OrderHistory';
// import Role from './components/market/usermanagement/Role';

// revamped sales
import SalesProduct from './components/sales/product/Product';
import SalesProductView from './components/sales/product/ProductView';
import SalesCustomer from './components/sales/customer/Customer';
import SalesCustomerView from './components/sales/customer/CustomerView';
import SalesSalesperson from './components/sales/salesperson/Salesperson';
import SalesSalespersonView from './components/sales/salesperson/SalespersonView';
import SalesOrder from './components/sales/ordermanagement/Order';
import SalesOrderView from './components/sales/ordermanagement/OrderView';
import Distributor from './components/sales/distributor/Distributor';
import DistributorView from './components/sales/distributor/DistributorView';

// Common Components
import SymphonySystemDialog from './components/symphony/SymphonySystemDialog';
// reducers and types
import { SystemState } from './store/system/types';
import { setSystemState, resetAxiosInterceptors } from './store/system/actions';

import { LoginState } from './store/login/types';
import SymphonyTokenLogin from './components/symphony/SymphonyTokenLogin';

import Geocode from 'react-geocode';
// constants
const GOOGLE_LIBRARIES = ['places'] as Libraries;
Geocode.setApiKey('AIzaSyB51sSAHDR8d73izjtJNCFMmZ6Zplqeqvw');
Geocode.setLanguage("en");
Geocode.setLocationType("ROOFTOP");

interface AppProps {
    setSystemState: typeof setSystemState;
    resetLoginState: typeof resetLoginState;
    setLoginState: typeof setLoginState;
    resetAxiosInterceptors: typeof resetAxiosInterceptors;
    login: LoginState;
    system: SystemState;
}

class App extends React.Component<AppProps> {
    node: Node = Object.create(HTMLElement.prototype, {})
    componentDidMount = () => {
        // Load Font Awesome
        this.node = loadCSS(
            'https://use.fontawesome.com/releases/v5.12.0/css/all.css',
            document.querySelector('#font-awesome-css'),
        );
        const { user } = this.props.login;
        const missingUser = typeof user.firstName === 'undefined' && typeof user.lastName === 'undefined';
        if (typeof this.props.login.token.token === 'undefined' || missingUser) {
            this.props.resetLoginState()
        } 
        if (!missingUser && typeof this.props.login.token.token !== 'undefined' ) {
            this.props.resetAxiosInterceptors();
        }
    }

    componentWillUnmount = () => {
        this.node.parentNode!.removeChild(this.node);
    }

    _onHideSystemDialog = () => {
        this.props.setSystemState({ 
            systemDialogOpen: false
        });
    }

    render () {
        const { shallRedirect, redirectTo } = this.props.system;
        const { isLoggedIn } = this.props.login;
        return (
            <Box className="App" width={1}>
                <LoadScript
                    googleMapsApiKey="AIzaSyBdyauj-IAvCaOQSwlGF4LEoNVhCGZD_Rs"
                    libraries={GOOGLE_LIBRARIES}
                >
                    <MuiPickersUtilsProvider utils={MomentUtils}>
                        <Router>
                            {isLoggedIn && <SymphonyDrawer />}
                            {isLoggedIn && <SymphonyHeader />}
                            {shallRedirect && redirectTo && <Redirect to={redirectTo} />}
                            <Switch>
                                <Route path="/login" exact={true} component={SymphonyLogin} />
                                <Route path="/login/:token" component={SymphonyTokenLogin} />
                                {/* <Route path="/basic/login" component={BasicLogin} /> */}
                                <Route path="/forgotpassword" component={SymphonyForgotPassword} />
                                <Route path="/newpassword" component={SymphonyNewPassword} />
                                {/* Revamped Symphony */}
                                <AuthenticatedRoute path="/home" component={Home} />
                                <AuthenticatedRoute path="/fields" component={Fields} />
                                <AuthenticatedRoute exact={true} path="/symphony/roles" component={Role} />
                                <AuthenticatedRoute path="/symphony/roles/:roleId" component={Accessibility} />
                                {/* Revamped Market */}
                                <AuthenticatedRoute exact={true} path="/market/product" component={MarketProduct} />
                                <AuthenticatedRoute path="/market/product/:productId" component={MarketProductView} />
                                <AuthenticatedRoute path="/market/review" component={ProductReview} />
                                <AuthenticatedRoute path="/market/rating" component={ProductRating} />
                                <AuthenticatedRoute exact={true} path="/market/promobanner" component={PromoBanner} />
                                <AuthenticatedRoute path="/market/promobanner/:bannerNumber" component={PromoBannerView}/>
                                <AuthenticatedRoute exact={true} path="/market/order" component={Order} />
                                <AuthenticatedRoute path="/market/order/:orderId" component={OrderView} />
                                <AuthenticatedRoute key="orderhistory" path="/market/orderhistory" component={OrderHistory} />
                                <AuthenticatedRoute path="/market/category" component={Category} />
                                <AuthenticatedRoute path="/market/aboutus" component={AboutUs} />
                                <AuthenticatedRoute path="/market/faqs" component={Faqs} />
                                <AuthenticatedRoute path="/market/helpdesk" component={HelpDesk} />
                                <AuthenticatedRoute exact={true} path="/market/vendor" component={Vendor} />
                                <AuthenticatedRoute exact={true} path="/market/useraccount" component={UserAccount} />
                                <AuthenticatedRoute path="/market/vendor/:vendorId" component={VendorView} />
                                <AuthenticatedRoute exact={true} path="/market/customer" component={Customer} />
                                <AuthenticatedRoute exact={true} path="/market/customer/:customerId" component={CustomerView} />
                                <AuthenticatedRoute exact={true} path="/symphony/role" component={Role} />
                                {/* Revamped Sales */}
                                <AuthenticatedRoute exact={true} path="/sales/salesperson" component={SalesSalesperson} />
                                <AuthenticatedRoute path="/sales/salesperson/:salespersonId" component={SalesSalespersonView} />
                                <AuthenticatedRoute exact={true} path="/sales/order" component={SalesOrder} />
                                <AuthenticatedRoute path="/sales/order/:orderId" component={SalesOrderView} />
                                <AuthenticatedRoute exact={true} path="/sales/distributor" component={Distributor} />
                                <AuthenticatedRoute path="/sales/distributor/:distributorId" component={DistributorView} />
                                <AuthenticatedRoute
                                    key="user-management"
                                    path="/Administration/Administration"
                                    component={UserManagement}
                                />
                                {/* Basic Routes */}
                                <AuthenticatedRoute
                                    key="calltypesettings"
                                    path="/Basic/CallTypeSettings/view"
                                    component={CallTypeSettings}
                                />
                                <AuthenticatedRoute
                                    key="applicationsettings"
                                    path="/basic/application/settings"
                                    component={ApplicationSettings}
                                />
                                <AuthenticatedRoute
                                    key="basicpromotions"
                                    path="/basic/promotions"
                                    component={Promotion}
                                />
                                <AuthenticatedRoute
                                    key="basicnews"
                                    path="/basic/news"
                                    component={News}
                                />
                                <AuthenticatedRoute
                                    key="accessibilitysettings"
                                    path="/basic/accessibilitysettings"
                                    component={AccessibilitySettings}
                                />
                                <AuthenticatedRoute
                                    key="basicnews"
                                    path="/basic/report"
                                    component={Report}
                                />
                                <AuthenticatedRoute
                                    exact={true}
                                    key="salesproduct"
                                    path="/sales/product"
                                    component={SalesProduct}
                                />
                                <AuthenticatedRoute
                                    key="salesproductview"
                                    path="/sales/product/:productId"
                                    component={SalesProductView}
                                />
                                <AuthenticatedRoute
                                    exact={true}
                                    key="salescustomer"
                                    path="/sales/customer"
                                    component={SalesCustomer}
                                />
                                <AuthenticatedRoute
                                    key="salescustomerview"
                                    path="/sales/customer/:customerId"
                                    component={SalesCustomerView}
                                />
                                <Route path="/">
                                    <Redirect to="/login" />
                                </Route>
                            </Switch>
                        </Router>
                        <SymphonySystemDialog
                            visible={this.props.system.systemDialogOpen}
                            onCloseAction={this._onHideSystemDialog.bind(this)}
                            maxWidth={this.props.system.systemDialogMaxWidth}
                            simpleDialog={this.props.system.systemDialogSimple}
                            content={this.props.system.systemDialogContent}
                            title={this.props.system.systemDialogTitle}
                            action={this.props.system.systemDialogActions}
                            simpleConfirm={this.props.system.systemDialogSimple}
                            onConfirmAction={this.props.system.systemDialogConfirmAction}
                            overrideTitle={this.props.system.systemOverrideTitle}
                            confirmOnly={this.props.system.systemConfirmOnly}
                        />
                    </MuiPickersUtilsProvider>
                </LoadScript>
            </Box>
        );
    }
}

const mapStateToProps = (state: AppState) => ({
    login: state.login,
    system: state.system
})

const mapDispatchToProps = { refreshToken, setHeaderText, resetLoginState, setLoginState, setSystemState, resetAxiosInterceptors };

export default connect(mapStateToProps, mapDispatchToProps)(App);
