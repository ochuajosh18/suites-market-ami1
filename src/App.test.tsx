import React from 'react';
import { mount } from 'enzyme';
import { createStore, combineReducers, applyMiddleware } from 'redux';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk'
import system from './store/system/reducers';
import login from './store/login/reducers';
import App from './App';

const createTestStore = () => {
    return createStore(
        combineReducers({
            login,
            system
        }),
        applyMiddleware(thunk)
    );
}

const setup = (store: ReturnType<typeof createTestStore>) => {
    let wrapper = mount(
        <Provider store={store}>
            <App />
        </Provider>
    )
    return wrapper;
}

jest.mock('fg-loadcss', () => ({
    loadCSS: jest.fn()
}));

jest.mock('@react-google-maps/api', () => ({
    LoadScript: (props) => <div>{props.children}</div>,
}));

describe('App', () => {
    let store: ReturnType<typeof createTestStore>;
    
    it('Renders the router and the app snapshot', () => {
        store = createTestStore();
        const wrapper = setup(store);
        expect(wrapper.render().html()).toMatchSnapshot()
    });
});