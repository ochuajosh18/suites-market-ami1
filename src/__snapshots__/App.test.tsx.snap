// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders the router and the app snapshot 1`] = `"<div><div class=\\"MuiBox-root MuiBox-root-3 Styled(MuiBox)-root-2\\" classes=\\"[object Object]\\"><div class=\\"MuiBox-root MuiBox-root-5 Styled(MuiBox)-root-4\\" classes=\\"[object Object]\\"><img alt=\\"\\" src=\\"symphony-logo-gold.png\\"></div><div class=\\"MuiBox-root MuiBox-root-7 Styled(MuiBox)-root-6\\" classes=\\"[object Object]\\"><div class=\\"MuiBox-root MuiBox-root-9 Styled(MuiBox)-root-8\\" classes=\\"[object Object]\\"><p class=\\"MuiTypography-root WithStyles(ForwardRef(Typography))-root-10 MuiTypography-body1\\">Welcome Back</p><div class=\\"MuiGrid-root WithStyles(ForwardRef(Grid))-root-12 MuiGrid-container\\" style=\\"margin-bottom: 20px;\\"><div class=\\"MuiGrid-root WithStyles(ForwardRef(Grid))-root-13 symphony-input-label MuiGrid-item MuiGrid-grid-xs-12\\">Platform</div><div class=\\"MuiGrid-root WithStyles(ForwardRef(Grid))-root-14 MuiGrid-item MuiGrid-grid-xs-12\\" style=\\"padding-top: 6px;\\"><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline\\"><div class=\\"MuiSelect-root WithStyles(ForwardRef(Select))-root-15 MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input\\" tabindex=\\"0\\" role=\\"button\\" aria-haspopup=\\"listbox\\" aria-label=\\"With out label\\" aria-labelledby=\\"symphony-login-platform-select\\" id=\\"symphony-login-platform-select\\">Symphony Market</div><input aria-hidden=\\"true\\" tabindex=\\"-1\\" class=\\"MuiSelect-nativeInput\\" value=\\"Symphony Market\\"><svg class=\\"MuiSvgIcon-root MuiSelect-icon\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M7 10l5 5 5-5z\\"></path></svg></div></div></div><div class=\\"MuiGrid-root WithStyles(ForwardRef(Grid))-root-12 MuiGrid-container\\" style=\\"margin-bottom: 20px;\\"><div class=\\"MuiGrid-root WithStyles(ForwardRef(Grid))-root-13 symphony-input-label MuiGrid-item MuiGrid-grid-xs-12\\">Email Address</div><div class=\\"MuiGrid-root WithStyles(ForwardRef(Grid))-root-14 MuiGrid-item MuiGrid-grid-xs-12\\" style=\\"padding-top: 6px;\\"><div class=\\"MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-16 MuiFormControl-fullWidth\\"><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" autocomplete=\\"none\\" id=\\"symphony-login-email-input\\" type=\\"text\\" maxlength=\\"999\\" class=\\"MuiInputBase-input MuiInput-input\\" value=\\"\\"></div></div></div></div><div class=\\"MuiGrid-root WithStyles(ForwardRef(Grid))-root-12 MuiGrid-container\\" style=\\"margin-bottom: 20px;\\"><div class=\\"MuiGrid-root WithStyles(ForwardRef(Grid))-root-13 symphony-input-label MuiGrid-item MuiGrid-grid-xs-12\\">Password</div><div class=\\"MuiGrid-root WithStyles(ForwardRef(Grid))-root-14 MuiGrid-item MuiGrid-grid-xs-12\\" style=\\"padding-top: 6px;\\"><div class=\\"MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-16 MuiFormControl-fullWidth\\"><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd\\"><input aria-invalid=\\"false\\" autocomplete=\\"new-password\\" id=\\"symphony-login-password-input\\" type=\\"password\\" class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd\\" value=\\"\\"><div class=\\"MuiInputAdornment-root MuiInputAdornment-positionEnd\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z\\"></path></svg></div></div></div></div></div></div><button class=\\"MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-17 MuiButton-text Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\" id=\\"symphony-login-button\\"><span class=\\"MuiButton-label\\">Login</span></button><div class=\\"MuiBox-root MuiBox-root-19 Styled(MuiBox)-root-18\\" classes=\\"[object Object]\\">Forgot your password?<button class=\\"MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-20 MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\" id=\\"symphony-forgetpassword-button\\"><span class=\\"MuiButton-label\\">Reset Password</span></button></div></div><div class=\\"Toastify\\"></div></div></div>"`;
